plasmapy/formulary/parameters.py

```diff
@@ -507,15 +536,43 @@ def thermal_speed(
     """
     m = mass if np.isfinite(mass) else particles.particle_mass(particle)
 
+    # TODO: Could implement this more cleanly for any integer ndim
+    # at the cost of using the gamma function in the expression for the
+    # mean magnitude velocity.
+
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
-    if method == "most_probable":
-        V = np.sqrt(2 * k_B * T / m)
-    elif method == "rms":
-        V = np.sqrt(3 * k_B * T / m)
-    elif method == "mean_magnitude":
-        V = np.sqrt(8 * k_B * T / (m * np.pi))
+    if ndim == 1:
+        if method == "most_probable":
+            V = 0 * u.m/u.s
+        elif method == "rms":
+            V = np.sqrt(k_B * T / m)
+        elif method == "mean_magnitude":
+            V = np.sqrt(2 * k_B * T / (m * np.pi))
+        else:
+            raise ValueError("Method {method} not supported in thermal_speed")
+
+    elif ndim == 2:
+        if method == "most_probable":
+            V = np.sqrt(k_B * T / m)
+        elif method == "rms":
+            V = np.sqrt(2 * k_B * T / m)
+        elif method == "mean_magnitude":
+            V = np.sqrt(np.pi * k_B * T / (2 * m))
+        else:
+            raise ValueError("Method {method} not supported in thermal_speed")
+
+    elif ndim == 3:
+        if method == "most_probable":
+            V = np.sqrt(2 * k_B * T / m)
+        elif method == "rms":
+            V = np.sqrt(3 * k_B * T / m)
+        elif method == "mean_magnitude":
+            V = np.sqrt(8 * k_B * T / (m * np.pi))
+        else:
+            raise ValueError("Method {method} not supported in thermal_speed")
     else:
-        raise ValueError("Method {method} not supported in thermal_speed")
+        raise ValueError("{ndim} is not a supported value for ndim in "
+                         "thermal_speed")
```

Comment:
Actually, you know what a nice refactor could be? A dictionary!

```python
coefficients = {1: {"most_probable": 0, ...}, ...}
try:
   coefficient = coefficients[ndim][method]
except KeyError:
    raise ValueError("Method {method} not supported in thermal_speed") 
    # could even break this down into two dict-indexing steps to handle both the exceptions
V = np.sqrt(coefficient * k_B * T / m)
```

The dictionary could even be created outside the function to save some completely unnoticeable amount of time on instantiating it on each call, if we care about that.

Reply:
Well that's a much, much better way to do that! Done.